From 3d1c9a76e7f0147feb65bd627583205d3a719ed0 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <chad.dupuis@cavium.com>
Date: Wed, 12 Oct 2016 11:32:47 -0400
Subject: [PATCH] fcoe: Allow more time for the bnx2x link negotiation before
 brining up fcoe interfaces.

bnx2x can take no longer than 3 seconds to initialize the link in some setups
which can cause fipvlan to fail and thus the fcoe interface(s) won't be
created.

Add another 10 seconds to give the link enough time to initialize.

Signed-off-by: Chad Dupuis <chad.dupuis@cavium.com>

Cherry-picked from: 3966a1e1ee0e3d27197258f446f54b683c415208
Resolves: #1378910
---
 modules.d/95fcoe/fcoe-up.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules.d/95fcoe/fcoe-up.sh b/modules.d/95fcoe/fcoe-up.sh
index cdff6ef1..a1f04747 100755
--- a/modules.d/95fcoe/fcoe-up.sh
+++ b/modules.d/95fcoe/fcoe-up.sh
@@ -87,8 +87,8 @@ elif [ "$netdriver" = "bnx2x" ]; then
     # If driver is bnx2x, do not use /sys/module/fcoe/parameters/create but fipvlan
     modprobe 8021q
     udevadm settle --timeout=30
-    # Sleep for 3 s to allow dcb negotiation
-    sleep 3
+    # Sleep for 13 s to allow dcb negotiation
+    sleep 13
     fipvlan "$netif" -c -s
 else
     vlan="no"
